src/elements/fields/PhoneField/index.tsx

```diff
@@ -0,0 +1,324 @@
+
+  useEffect(() => {
+    if (fullNumber === curFullNumber) return;
+
+    const ayt = new global.libphonenumber.AsYouType();
+    ayt.input(`+${fullNumber}`);
+    const numberObj = ayt.getNumber();
+    if (numberObj) {
+      setRawNumber(numberObj.nationalNumber);
+      setFormattedNumber(numberObj.formatNational());
+      setCurCountryCode(numberObj.country ?? DEFAULT_COUNTRY);
+    }
+  }, [fullNumber]);
+
+  useEffect(() => {
+    function hideOnClickAway(event: any) {
+      const clickedWithin = [triggerRef, dropdownRef].some((ref: any) =>
+        ref.current?.contains(event.target)
+      );
+      if (!clickedWithin) setShow(false);
+    }
+    document.addEventListener('mousedown', hideOnClickAway);
+    return () => document.removeEventListener('mousedown', hideOnClickAway);
+  }, []);
+
+  useEffect(() => {
+    if (element.properties.placeholder) return;
+
+    const exampleNumber = exampleNumbers[curCountryCode];
+    setPlaceholder(
+      global.libphonenumber
+        .parsePhoneNumber(exampleNumber, curCountryCode)
+        .formatNational()
+    );
+  }, [curCountryCode, element]);
+
+  useEffect(() => {
+    const phoneCode = countryMap[curCountryCode].phoneCode;
+    const newNumber = `${phoneCode}${rawNumber}`;
+    if ((fullNumber || rawNumber) && newNumber !== fullNumber) {
+      setCurFullNumber(newNumber);
+      onChange(newNumber);
+    }
+  }, [onChange, triggerOnChange, curCountryCode]);
+
+  const supportedCountries = useMemo(
+    () => (
+      <>
+        <input
+          autoFocus
+          placeholder='Search'
+          css={{
+            width: '100%',
+            height: '30px',
+            outline: 'none',
+            border: 'none',
+            borderBottom: '1px solid grey',
+            padding: '5px 10px',
+            fontSize: '14px'
+          }}
+          onChange={(e) => setQuery(e.target.value.toLowerCase())}
+        />
+        {countryData
+          .filter(({ countryCode, countryName, phoneCode }) => {
+            if (!global.libphonenumber.isSupportedCountry(countryCode))
+              return false;
+            return (
+              !query ||
+              countryCode.toLowerCase().includes(query) ||
+              countryName.toLowerCase().includes(query) ||
+              `+${phoneCode}`.includes(query)
+            );
+          })
+          .map(({ flag, countryCode, countryName, phoneCode }) => (
+            <li
+              key={countryCode}
+              css={{
+                padding: '2px 10px',
+                display: 'flex',
+                alignItems: 'center',
+                transition: '0.1s ease all',
+                fontSize: '14px',
+                '&:hover': {
+                  backgroundColor: '#e6e6e6'
+                }
+              }}
+              onClick={() => {
+                setCurCountryCode(countryCode);
+                setRawNumber('');
+                setFormattedNumber('');
+                setShow(false);
```

Comment:
After changing countries, it would be good to put the focus on the input field for immediate entry of a number.